packages/core/src/rules/oas3/no-invalid-media-type-examples.ts

```diff
@@ -13,35 +13,42 @@ export const ValidContentExamples: Oas3Rule = (opts) => {
         const { location, resolve } = ctx;
         if (!mediaType.schema) return;
         if (mediaType.example !== undefined) {
-          resolveAndValidateExample(mediaType.example, location.child('example'));
+          resolveAndValidateExample(mediaType.example, location.child('example'), location);
         } else if (mediaType.examples) {
           for (const exampleName of Object.keys(mediaType.examples)) {
             resolveAndValidateExample(
               mediaType.examples[exampleName],
               location.child(['examples', exampleName, 'value']),
+              location,
               true
             );
           }
         }
 
         function resolveAndValidateExample(
           example: Oas3Example | any,
-          location: Location,
+          exampleLocation: Location,
+          mediaTypeLocation: Location,
           isMultiple?: boolean
         ) {
           if (isRef(example)) {
             const resolved = resolve<Oas3Example>(example);
             if (!resolved.location) return;
-            location = isMultiple ? resolved.location.child('value') : resolved.location;
+            exampleLocation = isMultiple ? resolved.location.child('value') : resolved.location;
             example = resolved.node;
           }
           if (isMultiple && typeof example?.value === 'undefined') {
             return;
           }
+
+          const resolvedSchema = ensureSchemaIsResolved(mediaType.schema!, ctx);
```

Comment:
I guess it will still be easier to fix that on the AJV side. We use our own AJV fork, and we can try passing down a new parameter, something like `ignoreWriteOnly`. Let me think this through.